components/Players/PlayerManager.vue

```diff
@@ -195,6 +203,34 @@ export default Vue.extend({
         default:
           return this.getCurrentItem.Name;
       }
+    },
+    timeLeft(): number {
+      return parseInt((this.mediaDuration - this.getCurrentTime).toFixed());
+    },
+    mediaDuration(): number {
+      // In seconds
+      return this.ticksToMs(this.getCurrentItem.RunTimeTicks) / 1000;
+    },
+    showUpNext(): boolean {
+      if (
+        this.isMinimized ||
+        this.upNextUserHidden ||
+        this.getCurrentItem.Type !== 'Episode' ||
+        !this.getNextItem
+      ) {
+        return false;
+      }
+      // How much of the video to go through before showing
+      // the 'up-next' component
+      const showAtProgressPercentage = 0.97;
```

Comment:
Percentages for this kind of thing don't really work well. It should use a fixed duration.

Some examples:

* For a 2h video, this would be ~3min30
* For a 10min show, this would be ~18 seconds
* For a 5h video, this would be ~9min

While not as extreme as the same issues we've had on in general with Audiobooks progress using percentages to mark things as played, this doesn't scale very well, imo.

We could probably use like 30 seconds for now, then once we have proper intro/outro detection, rework this to also take that into account.

Reply:
To me 3min30 in a 2h media doesn't seem too far fetched, as it's generally a movie with credits.

Reply:
For a miniseries (Battlestar Galactica for example) you'd probably be eating a few of the final minutes though.

Reply:
Imo there's no universal solution until we get the nice intro/outro detection system. So I'd say we shouldn't worry too much, especially when the "next up" prompt is discardable. So either a fixed timing or a percentage is fine to me

Reply:
The Kodi Next up plugin have configurable options. It shows the popup 45 seconds before the end on a 45 minute video for example. there is also the option for <=15 minute videos and 1h30+ videos (not sure but you get the idea) so it's more tunable depending on the "general context".

Reply:
That seems like a good compromise, yeah.

We can set defaults based on these lengths now, then provide configuration when we get to config pages.